mypyc/genops.py

```diff
@@ -266,6 +266,54 @@ def visit_for_stmt(self, s: ForStmt) -> int:
 
             self.pop_loop_stack(end_block, next)
             return -1
+
+        if self.node_type(s.expr).name == 'list':
+            self.push_loop_stack()
+
+            expr_reg = self.accept(s.expr)
+
+            index_reg = self.alloc_temp(RTType('int'))
+            self.add(LoadInt(index_reg, 0))
+
+            one_reg = self.alloc_temp(RTType('int'))
+            self.add(LoadInt(one_reg, 1))
+            
+            assert isinstance(s.index, NameExpr)
+            assert isinstance(s.index.node, Var)
+            lvalue_reg = self.environment.add_local(s.index.node, self.node_type(s.index))
+
+            len_reg = self.alloc_temp(RTType('int'))
+            self.add(PrimitiveOp(len_reg, PrimitiveOp.LIST_LEN, expr_reg))
+
+            condition_block = self.goto_new_block()
+            branch = Branch(index_reg, len_reg, -1, -1, Branch.INT_LT)
+            self.add(branch)
+            branches = [branch]
+            
+            body_block = self.new_block()
+            self.set_branches(branches, True, body_block)
+
+            target_list_type = self.types[s.expr]
+            assert isinstance(target_list_type, Instance)
+            target_type = type_to_rttype(target_list_type.args[0])
+            value_box = self.alloc_temp(RTType('int'))
```

Comment:
The type should be `RTType('object')` since a Python list object can have arbitrary item types. The unboxing operation will perform the necessary runtime type check.